spark3/src/main/java/org/apache/iceberg/spark/source/SparkWrite.java

```diff
@@ -206,9 +215,7 @@ private void abort(WriterCommitMessage[] messages) {
             PropertyUtil.propertyAsInt(props, COMMIT_TOTAL_RETRY_TIME_MS, COMMIT_TOTAL_RETRY_TIME_MS_DEFAULT),
             2.0 /* exponential */)
         .throwFailureWhenFinished()
-        .run(file -> {
-          io.value().deleteFile(file.path().toString());
-        });
+        .run(file -> table.io().deleteFile(file.path().toString()));
```

Comment:
Nit: non-functional change that may cause commit conflicts.

Reply:
Reverted.